include/boost/geometry/algorithms/detail/is_valid/polyhedral_surface.hpp

```diff
@@ -0,0 +1,462 @@
+// Boost.Geometry
+
+// Copyright (c) 2025 Oracle and/or its affiliates.
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_IS_VALID_POLYHEDRAL_SURFACE_HPP
+
+#include <boost/qvm/vec.hpp>
+#include <boost/qvm/vec_operations.hpp>
+
+#include <boost/geometry/algorithms/detail/is_valid/interface.hpp>
+#include <boost/geometry/algorithms/detail/is_valid/polygon.hpp>
+#include <boost/geometry/algorithms/correct.hpp>
+#include <boost/geometry/io/wkt/wkt.hpp>
+#include <boost/geometry/strategies/transform/matrix_transformers.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace is_valid
+{
+
+template <typename PolyhedralSurface, bool AllowInconsistentOrientation = false>
+struct valid_intersection
+{
+    valid_intersection() = default;
+
+    using CT = typename coordinate_type<PolyhedralSurface>::type;
+    using vec3 = boost::qvm::vec<CT, 3>;
+    using polygon_3d = typename boost::range_value<PolyhedralSurface>::type;
+    using ring_3d = typename boost::geometry::ring_type<polygon_3d>::type;
+    using point_3d = typename boost::range_value<ring_3d>::type;
+    using segment_3d = typename boost::geometry::model::segment<point_3d>;
+    using set_vec_from_point = typename strategy::transform::detail::matrix_transformer::
+        set_vec_from_point<point_3d, 0, 3>;
+
+    using point_2d = boost::geometry::model::point
+        <CT, 2, boost::geometry::cs::cartesian>;
```

Comment:
Are polyhedral surfaces always cartesian (I didn't see a corresponding constraint in the concept check https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/geometries/concepts/polyhedral_surface_concept.hpp but I may have missed it) and if not should this be generic or is_valid constrained for now?

Reply:
I think it's a valid constraint at least for now

Reply:
I added that as a concept constraint together with the constraint on dimension (==3).